keps/sig-network/3726-standard-application-protocols/README.md

```diff
@@ -0,0 +1,284 @@
+
+While this is a good solution - we never came with recommended standards other than [IANA standard service names](https://www.iana.org/assignments/service-names) or a domain prefixed protocol.
+
+This loose definition has led us to;
+1. Have instances where implementations do different things for common protocols.
+2. Have no support for implementations interoperability with different domain prefixed protocols (or a mix domain prefixed and non prefixed protocol) for the same port.
+
+One good example for the first bullet is `HTTP2`. 
+* In GKE you can use `appProtocol: HTTP2` and it will describe HTTP2 over TLS (https://cloud.google.com/kubernetes-engine/docs/how-to/secure-gateway#load-balancer-tls).
+* While in Istio it will be h2c (HTTP2 over cleartext).
+
+That creates a problem where users with GKE and Istio in their cluster can have very different behaviors for the same `appProtocol` value.
+
+
+### Goals
+
+* Rephrase AppProtocol field description
+* Build consensus around how common (non IANA service names) should be implemented
+* Help the broader community and specifically implementations to interoperate better
+* Provide short and clear documentation for how AppProtocol values should be implemented
+  * Update the appProtocol user documentation respectively
+
+
+### Non-Goals
+
+* Validate appProtocol values
+* Monitor appProtocol implementations
+* Support multiple AppProtocols values for the same port to improve interoperability (suggested as a followup work though)
+
+
+## Proposal
+
+~~Kubernetes supports the `appProtocol` field to provide a way to specify an application protocol for each Service port.~~
+
+Kubernetes `appProtocol` field is used as a hint for implementations to configure the protocol used between the implementation and the application it exposes.
+
+The [documentation](https://kubernetes.io/docs/concepts/services-networking/service/#application-protocol) for this field says that:
+
+```Values should either be IANA standard service names or domain prefixed names such as mycompany.com/my-custom-protocol.```
+
+This KEP proposes to declare standard Kubernetes protocol names for common protocols that are not IANA standard service names.
+
+Those common protocols will be well defined strings prefixed with ‘k8s.io’. 
+`k8s.io/h2c` as an example.
+
+### list of protocols
+- 'k8s.io/http2'
+- 'k8s.io/grpc'
+- 'k8s.io/tcp'
+
+### Risks and Mitigations
+
+There are no real “risks”, primary concerns are:
+1. End users will not migrate to `k8s.io/<>` values
+2. It will take long time to implementations align with the new standards
+3. We have no easy way to monitor who is aligned and who is not
+
+
+## Design Details
+
+At first, the list is going to live in `ServicePort` and `EndpointPort` as part of the AppProtocol description.
```

Comment:
What is "the list" ?

Reply:
Oh, the list of protocols?  Let's jusmp straight to metadata.

Reply:
I have changed the wording a bit